lib/sheet.js

```diff
@@ -0,0 +1,76 @@
+import Spreadsheet from 'edit-google-spreadsheet';
+import googleAuth from './google-auth';
+import util from './util';
+
+let SHEET_CONFIG;
+util.readJsonFromFile('./google-spreadsheet-config.json', data => {
+    SHEET_CONFIG = data;
+});
+
+class Sheet {
+    constructor() {
+        this.dataCache = {};
+        return this;
+    }
+
+    fetch(searchString, processingCallback, afterRequestCallback){
+        if (!this.dataCache.extensions) {
+            this.updateCache(searchString, processingCallback, afterRequestCallback);
+        }else{
+            processingCallback(searchString, this.dataCache, afterRequestCallback);
+        }
+    }
+
+    updateCache(searchString, processingCallback, afterRequestCallback){
+        googleAuth.authenticate(tokens => {
+            this.requestSpreadsheet(SHEET_CONFIG.extensionsWorksheetId, SHEET_CONFIG.spreadsheetId, tokens, (extensions) => {
+                this.requestSpreadsheet(SHEET_CONFIG.aliasesWorksheetId, SHEET_CONFIG.spreadsheetId, tokens, (aliases) => {
+                    this.dataCache = this.buildData(extensions, aliases);
+                    if (processingCallback){
+                        processingCallback(searchString, this.dataCache, afterRequestCallback);
+                    }
+                });
+            });
+        });
+    }
+
+    requestSpreadsheet(worksheetId, spreadsheetId, tokens, callback){
```

Comment:
In that line of thought, then what we could do is use this Class as a real Class (instead of just returning a singleton instance of it). So other modules that would like to work with different spreadsheets would only have to instantiate it with other credentials. In other words: we move all the configuration settings for its constructor and later on we'd just use its properties (e.g.: `this.tokens`, `this.spreadsheetId`). What about it?